plugins/carbon_black_cloud/icon_carbon_black_cloud/actions/quarantine/action.py

```diff
@@ -15,30 +15,11 @@ def __init__(self):
         )
 
     def run(self, params={}):
-        agent = params.get(Input.AGENT)
-        whitelist = params.get(Input.WHITELIST)
+        # START INPUT BINDING - DO NOT REMOVE - ANY INPUTS BELOW WILL UPDATE WITH YOUR PLUGIN SPEC AFTER REGENERATION
+        agent = params.get(Input.AGENT, "")
+        whitelist = params.get(Input.WHITELIST, [])
         quarantine_state = params.get(Input.QUARANTINE_STATE)
-
-        if quarantine_state and whitelist_checker.match_whitelist(agent, whitelist, self.logger):
-            self.logger.info(f"Agent {agent} matched item in whitelist, skipping quarantine.")
-            return {Output.QUARANTINED: False}
-
-        agent_object = self.connection.get_agent(agent)
-        agent_id = agent_object.get("id")
-
-        if quarantine_state:
-            toggle = "ON"
-        else:
-            toggle = "OFF"
-
-        payload = {
-            "action_type": "QUARANTINE",
-            "device_id": [str(agent_id)],
-            "options": {"toggle": toggle},
-        }
-
-        url = f"{self.connection.base_url}/appservices/v6/orgs/{self.connection.org_key}/device_actions"
-        self.connection.post_to_api(url, payload)
+        # END INPUT BINDING - DO NOT REMOVE
 
         # This API returns 204 no content if successful, we have to assume the state was applied on a successful call
-        return {Output.QUARANTINED: quarantine_state}
+        return {Output.QUARANTINED: self.connection.update_quarantine_state(agent, whitelist, quarantine_state)}
```

Comment:
Would it be better to separate this logic just a little bit?
```suggestion
        response = self.connection.update_quarantine_state(agent, whitelist, quarantine_state)
        return {Output.QUARANTINED: response}
```